### ISSUE TITLE
Samplesheets: Add UI for configuring field editing

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-server#698]
To be done after #696. The sheet configs are getting too complex to nicely edit by hand, even during development.

I will implement a simple modal menu for configuring valid values in a column, if user has sheet management access.

Functionality to be added as we implement more stuff (e.g. ontology and list support)

## Temp Tasks

- [x] Disable editing if type == 'select' but no options are given
- [x] Apply config changes into UI
- [x] Dynamically enable/disable cell editor
- [x] Update study column config changes on assay (and vice versa)
- [x] Fix modal content updating if `editable==false`
- [x] Fix default regex activation if modifying column format
- [x] Fix disabling update button with initial null range values
- [x] Move multi-table config and editor updating from main app to `ManageColumnModal`
- [x] Save config changes on server
- [x] ~~Fix "editable" status updating on column toggle modal~~ See #710 and #711
- [x] Add column config clipboard copy/paste
- [x] Remove `type` and `name` from clipboard copying
- [x] ~~Add CTRL+C -copying for full data~~ Nope, too many side effects
- [x] ~~Add legend for full data copying~~ Also nope, add paste input instead
- [x] Add UI notifications
- [x] Add validity check for default value
- [x] Refactor validation and `setUpdateState()` to also work on full paste
- [x] Fix enabling config saving if format changes to integer and default value is not changed
- [x] Add default value/unit updating
- [x] Initialize config modal based on existing values (see "Age" in i_small.zip)
- [x] Fix config button visible for performer and perform date (see i_small.zip)
- [x] Update column rendering dynamically based on format (e.g. integer -> right align and show unit)
- [x] ~~Add tests~~ Postponed to v0.7.1, see #749
